lib/pbench/test/unit/server/test_indexing_tarballs.py

```diff
@@ -265,6 +266,41 @@ def error(self, dataset: Dataset, message: str):
         __class__.errors[dataset.name] = message
 
 
+class FakeLockManager:
+    def __init__(self, lock: Path, exclusive: bool = False, wait: bool = True):
+        """Initialize a mocked lock reference
+
+        Args:
+            lock: the path of a lock file
+            exclusive: lock exclusively
+            wait: wait for lock
+        """
+        self.exclusive = exclusive
+        self.wait = wait
+
+    def __enter__(self) -> "FakeLockManager":
+        """Acquire the lock
+
+        Returns:
+            self reference for 'as' clause
+        """
+        return self
+
+    def __exit__(self, *exc):
+        """Release the lock and close the lock file"""
+        self.exclusive = False
+
+    def upgrade(self):
+        """Upgrade a shared lock to exclusive"""
+        if not self.exclusive:
+            self.exclusive = True
+
+    def downgrade(self):
+        """Downgrade an exclusive lock to shared"""
+        if self.exclusive:
+            self.exclusive = False
```

Comment:
Just a nit, but I'm not sure that the conditionals are worthwhile:  we can just set the values unconditionally -- sometimes it will change the value and sometimes it will overwrite the value with itself -- but it'll produce the correct result in each case.  (We needed the conditionals in the CUT, because they protect other code, but here I think they probably just impose cost and complexity (albeit a _tiny_ amount) for what is probably no benefit.)

Reply:
I carefully considered removing them for almost 1/4 of a second when converting the fake `LockRef` to a fake `LockManager`, and decided that it wasn't worth the effort. I'll devote at most another 1/4 second to reconsidering this if you gave me some other reason to open the file, but no promises. 😆

Reply:
I ain't giving you no reason!  🙃